src/plots/linePlot.ts

```diff
@@ -425,6 +450,74 @@ export module Plots {
       return filteredDataIndices;
     }
 
+    private _filterDownsampling(dataset: Dataset, indices: number[]) {
+      let xAccessor = this.x().accessor;
+      let yAccessor = this.y().accessor;
+      let filteredIndices = this._filterDownsamplingSlope(dataset, indices, this.x().scale, this.y().scale, xAccessor, yAccessor);
+      return filteredIndices;
+    }
+
+    private _filterDownsamplingSlope(dataset: Dataset, indices: number[],
+      xScale: Scale<any, number>, yScale: Scale<any, number>, xAcessor: Accessor<any>, yAcessor: Accessor<any>) {
```

Comment:
is there any reason why you need to pass in scales and accessors here? they should be accessible via `this`